docs/interfaces.md

```diff
@@ -12,198 +12,362 @@ Slots may support multiple connections to plugs.  For example the OS snap
+``bluez`` interface is not provided by the OS snap so a snap author
+implementing the bluez service might use ``slots: [ bluez ]``. Then after
+install, the bluez interface shows up as available:
+
+    $ sudo snap install foo-blue
+    $ snap interfaces
+    Slot                 Plug
+    foo-blue:bluez       -
 
-Can access non-hidden files in user's `$HOME` to read/write/lock.
-This is restricted because it gives file access to the user's
-`$HOME`. This interface is auto-connected on classic systems and
-manually connected on non-classic.
+Now install and connect works like before (eg, ``baz`` uses
+``plugs: [ bluez ]``):
 
-Usage: reserved
-Auto-Connect: yes
+    $ sudo snap install baz
+    $ snap interfaces
+    Slot                 Plug
+    foo-blue:bluez       -
+    -                    baz:bluez
+    $ sudo snap connect baz:bluez foo-blue:bluez
+    $ snap interfaces
+    Slot                 Plug
+    foo-blue:bluez       baz:bluez
+
+## Supported Interfaces - Basic
+
+### camera
+
+Can access the first video camera. Suitable for programs wanting to use
+webcams.
+
+* Auto-Connect: no
+* Availability: OS snap (classic)
 
 ### gsettings
 
-Can access global gsettings of the user's session. This is restricted because
-it gives privileged access to sensitive information stored in gsettings and
-allows adjusting settings of other applications.
+Can access global gsettings of the user's session which gives privileged access
+to sensitive information stored in gsettings and allows adjusting settings of
+other applications.
 
-Usage: reserved
-Auto-Connect: yes
+* Auto-Connect: yes
+* Availability: OS snap (classic)
 
-### optical-drive
+### home
 
-Can access the first optical drive in read-only mode. Suitable for CD/DVD playback.
+Can access non-hidden files in user's `$HOME` and gvfs mounted directories
+owned by the user to read/write/lock.
 
-Usage: common
-Auto-Connect: yes
+* Auto-Connect: yes on classic, no on native
+* Availability: OS snap
 
 ### mpris
 
-Can access media players implementing the Media Player Remote Interfacing
-Specification (mpris) when the interface is specified as a plug.
+Providing snaps implementing the Media Player Remove Interfacing Specification
+(mpris) may be accessed via their well-known DBus name.
 
-Media players implementing mpris can be accessed by connected clients when
-specified as a slot.
+Consuming snaps can access media players implementing mpris via the providing
+snap's well-known DBus name.
 
-Usage: common
-Auto-Connect: no
+* Auto-Connect: no
+* Availability: with providing snap
 
-### camera
+### network
 
-Can access the first video camera. Suitable for programs wanting to use the
-webcams.
+Can access the network as a client.
+
+* Auto-Connect: yes
+* Availability: OS snap
+
+### network-bind
+
+Can access the network as a server.
+
+* Auto-Connect: yes
+* Availability: OS snap
+
+### opengl
+
+Can access OpenGL hardware.
+
+* Auto-Connect: yes
+* Availability: OS snap (classic)
```

Comment:
Why classic?  What if the device has a screen?  What if I install a native snap system on my laptop and install a unity8 snap?